data_structures/binary_search_tree.cpp

```diff
@@ -99,12 +99,14 @@ void BFT(node *n) {
 }
 
 void Pre(node *n) {
-    if (n != NULL) {
-        std::cout << n->val << "  ";
+    std::cout << ->val<< "  ";
```

Comment:
I'm not sure this is right: the previous code seemed fine, IMO.
@tjgurwara99, what are your thoughts on this?

Reply:
This is incorrect. This is exactly the reason why the CodeQL auto build is also failing

```
/home/runner/work/C-Plus-Plus/C-Plus-Plus/data_structures/binary_search_tree.cpp:102:18: error: expected primary-expression before ‘->’ token
```

For the use of arrow operator `->`, you need an identifier (in this case `n`) on the left hand side.